Raven.Studio.Html5/App/commands/database/documents/getCollectionsLabelsCommand.ts

```diff
@@ -0,0 +1,59 @@
+﻿import commandBase = require("commands/commandBase");
+import database = require("models/resources/database");
+import collection = require("models/database/documents/collection");
+import queryUtil = require("common/queryUtil");
+
+class getCollectionsLabelsCommand extends commandBase {
+
+    /**
+    * @param ownerDb The database the collections will belong to.
+    */
+    constructor(private collections: collection[], private ownerDb: database) {
+        super();
+
+        if (!this.ownerDb) {
+            throw new Error("Must specify a database.");
+        }
+    }
+
+
+    execute(): JQueryPromise<collection[]> {
+        var task = $.Deferred();
+
+        var requests = this.collections.map(collection => {
+            return {
+                Url: "/queries/",
+                Headers: {},
+                Query: "?id=" + queryUtil.escapeTerm("Raven/StudioConfig")
+            }
+        });
+        this.post("/multi_get?parallel=yes", JSON.stringify(requests), this.ownerDb, null, 0)
+            .done((result) => {
+                // if there are no results, then we simply need to leave and go back to normal collection display
+                if (!result[0].Result.Results[0])
+                    task.resolve(this.collections);
+                var arr = result[0].Result.Results[0].Labels;
+                for (var i = 0; i < this.collections.length; i++) {
+                    if (arr[this.collections[i].collectionName] !== undefined) {
+                        this.collections[i].collectionLabel =
+                            arr[this.collections[i].collectionName];
+                    } else {
+                        this.collections[i].collectionLabel = this.collections[i].collectionName;
+                    }
+                }
+                // sort the collections alphabetically
+                task.resolve(this.collections.sort((n, r) => {
+                    return (n.collectionLabel === r.collectionLabel) ? 0 : (n.collectionLabel > r.collectionLabel) ? 1 : -1;
```

Comment:
For single statement functions TypeScript offers you shorter syntax:
task.resolve(this.collections.sort((n, r) =>  (n.collectionLabel === r.collectionLabel) ? 0 : (n.collectionLabel > r.collectionLabel) ? 1 : -1));